src/main/java/jarvis/task/TaskList.java

```diff
@@ -0,0 +1,143 @@
+package jarvis.task;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import jarvis.exception.command.CommandParseException;
+import jarvis.exception.command.InvalidIndexException;
+import jarvis.exception.command.InvalidParameterException;
+
+/**
+ * Container class for tasks and their corresponding operations.
+ */
+public class TaskList {
+    private final List<Task> tasks;
+
+    /**
+     * Constructor for an empty task list.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+    }
+
+    /**
+     * Constructor for a task list populated with the given tasks.
+     *
+     * @param tasks Tasks to populate the list with.
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = new ArrayList<>(tasks);
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task Task to add.
+     * @return List of response lines.
+     */
+    public List<String> addTask(Task task) {
+        assert task != null;
+        this.tasks.add(task);
+        return List.of(
+                String.format("Got it! I've added task %d to the list.", this.tasks.size()),
+                "\t" + task
+        );
+    }
+
+    /**
+     * Deletes a task from the list at the given index.
+     *
+     * @param index Index of the task to delete.
+     * @return List of response lines.
+     * @throws InvalidIndexException If the index to delete is invalid.
+     */
+    public List<String> deleteTask(int index) throws CommandParseException {
+        if (this.tasks.isEmpty()) {
+            throw new InvalidParameterException(
+                    "TaskList is empty",
+                    "There are no tasks to delete, please add a task first."
+            );
+        }
+        if (index <= 0 || index > this.tasks.size()) {
+            throw new InvalidIndexException(1, this.tasks.size());
```

Comment:
Perhaps this `1` here can be interpreted as a magic number that may cause confusion. Can consider having it as a named constant to make it more descriptive!